src/app.module.ts

```diff
@@ -27,6 +27,29 @@ import { HealthCheckModule } from './health-check/health-check.module';
 import { DIDContactModule } from './modules/did-contact/did.contact.module';
 import { StatusListModule } from './modules/status-list/status-list.module';
 import { IPFSModule } from './modules/ipfs/ipfs.module';
+import { ValidationOptions } from 'joi';
+import { envVarsValidationSchema } from './env-vars-validation-schema';
+
+const validationOptions: ValidationOptions = {
+  stripUnknown: true,
+  allowUnknown: false,
+  abortEarly: false,
+};
+
+let config: DynamicModule;
+
+try {
```

Comment:
what is the reason of creating config object in try block? doesn't nests.js terminates app if confing invalid? will env variables be loaded outside of @Module?

Reply:
It allows exiting with a clean message. This is the only reason I have implemented it like this.

Reply:
It is 
```
Error: Config validation error: "PUBLIC_RESOLVER_ADDRESS" must be a valid ethereum address
exiting
```
vs
```
/Users/artur/WebstormProjects/energyweb/iam/ssi-hub/node_modules/@nestjs/config/dist/config.module.js:78
                throw new Error(`Config validation error: ${error.message}`);
                      ^
Error: Config validation error: "PUBLIC_RESOLVER_ADDRESS" must be a valid ethereum address
    at Function.forRoot (/Users/artur/WebstormProjects/energyweb/iam/ssi-hub/node_modules/@nestjs/config/dist/config.module.js:78:23)
    at Object.<anonymous> (/app.module.ts:63:18)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/main.ts:2:1)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
```